js/lib/index.js

```diff
@@ -0,0 +1,703 @@
+   *
+   * @since 1.0.0
+   */
+  getRecording (title) {
+    return _omiStateEntry(this._sawtoothRestUrl, Recording, title)
+  }
+
+  getRecordings () {
+    return _omiStateCursor(this._sawtoothRestUrl, Recording)
+  }
+
+  /**
+   * Set a Work.
+   *
+   * A SetWork transaction will be submitted to the blockchain and a URL for the
+   * resulting batch status will be returned via a promise.
+   *
+   * The argument should be structured like so:
+   *
+   * ```
+   * {
+   *   "title": "String", // a unique title of the work; the natural key
+   *   "ISWC": "International Standard Work Code", // may be null
+   *   "songwriter_publisher_splits": [
+   *     {
+   *       // all splits in this list must add up to 100
+   *       "split": <a positive int>,
+   *       "songwriter_publisher": {
+   *         "songwriter_name": "IndividualIdentity.name",
+   *         "publisher_name": "OrganizationalIdentity.name" // must be Type.PUBLISHER
+   *       }
+   *     },
+   *     // and more
+   *   ]
+   * }
+   * ```
+   *
+   * Only previously set objects using the same private key as this instance may
+   * be overwritten.
+   *
+   * @param {object} work - the Work to be set; see method doc for structure
+   *
+   * @returns {Promise<BatchStatus>} a promise for the result of transaction submission
+   *
+   * @since 1.0.0
+   */
+  setWork (work) {
+    return _submitOmiTransaction(this._sawtoothRestUrl,
+                                 this._privateKey,
+                                 'SetWork',
+                                 Work,
+                                 'title',
+                                 work)
+  }
+
+  /**
+   * Fetch a Work from the global state, and return the value via a promise.
+   *
+   * The resulting object returned on the resolved promis will be structured
+   * like so:
+   *
+   * ```
+   * {
+   *   "title": "String", // a unique title of the work; the natural key
+   *   "ISWC": "International Standard Work Code", // may be null
+   *   "songwriter_publisher_splits": [
+   *     {
+   *       // all splits in this list must add up to 100
+   *       "split": <a positive int>,
+   *       "songwriter_publisher": {
+   *         "songwriter_name": "IndividualIdentity.name",
+   *         "publisher_name": "OrganizationalIdentity.name" // must be Type.PUBLISHER
+   *       }
+   *     },
+   *     // and more
+   *   ]
+   * }
+   * ```
+   *
+   * @param {string} title - the title of the work (its natural key)
+   *
+   * @returns {Promise} the promise for the work value
+   *
+   * @since 1.0.0
+   */
+  getWork (title) {
+    return _omiStateEntry(this._sawtoothRestUrl, Work, title)
+  }
+
+  getWorks () {
+    return _omiStateCursor(this._sawtoothRestUrl, Work)
+  }
+}
+
+/**
+ * A BatchStatus object provides a function to query for the current batch
+ * status of a submitted transaction. That is, whether or not the batch has been
+ * committed to the block chain.
+ */
+class BatchStatus {
```

Comment:
I wonder if `BatchStatus` is the most appropriate name. Two thoughts come to mind:
1) The concept of _"batches"_ has mostly been abstracted away in this client. Is there a more OMI-specific object the name could refer to?
2) The name implies that it is itself the status, not a thing that checks the status.

Suggestions:
- `CommitStatus`
- `CommitChecker`
- `StatusChecker`